docs-src/tls.md

```diff
@@ -0,0 +1,328 @@
+# TLS
+
+Service APIs allow for a variety of ways to configure TLS. This document lays
+out various TLS settings and gives general guidelines on how to use them
+effectively.
+
+## Client/Server and TLS
+
+![overview](tls-overview.svg)
+
+For Gateways, there are two connections involved:
+
+- **Downstream**: This is the connection between the client and the Gateway.
+- **Upstream**: This is the connection between the Gateway and Service. The
+  Service in question can be a core Kubernetes Service or any other network
+  endpoint as well.
```

Comment:
This is really hard to describe well, I've never found a great way to say it, but maybe something like this?
```suggestion
- **Upstream**: This is the connection between the Gateway and backend resources
   specified by routes. These backend resources will usually be Services.
```